sdk/core/Azure.Core/src/LongRunningOperation.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure
+{
+    public abstract class LongRunningOperation<T>: IDisposable
+    {
+        public abstract Response Raw { get; }
+        public abstract ValueTask<Response<T>> GetValueAsync(CancellationToken cancellationToken = default);
```

Comment:
I am trying to keep it as a sync Value property. The way I designed it is you have to call "get status" at least once, and if the status is "completed", the value is read and deserialized inside the "get status call". If get status is not called before Value is accessed, the accessor throws.

Reply:
I'm afraid that `have to call "get status" at least once` not obvious from the API so people would have to get a runtime exception to understand how long-running API works.

Reply:
I'd like to most obvious thing to "just work"

Reply:
I think it would be nice to get parity with Response.
Also, your GetValueAsync method does not return value (i.e. the T)
Lastly, I was thinking that maybe we could unify Response<T> and Operation<T> if they have the same APIs for accessing the value and the raw response. The only difference between these two types would be that one needs to be "materialized" by calling some i/o methods.

Would you mind prototyping what that would look like?

Reply:
> Also, your GetValueAsync method does not return value (i.e. the T)

It returns `Response<T>` like every other method that does HTTP call and deserializes the value.

> Would you mind prototyping what that would look like?

I'm not sure I've understood the idea.

Reply:
> I'm not sure I've understood the idea.

Let me try to finish it in my prototype